### PR TITLE
Turn off Interrupt Triggers and Pull Up Resistors on begin()

### PR DESCRIPTION
**Pull request by PsuFan**
- **Describe the scope of your change--i.e. what the change does and what parts
  of the code were modified.**  
Turn off Interrupt Triggers and Pull Up Resistors on begin()

- **Describe any known limitations with your change.**  
None

- **Please run any tests or examples that can exercise your modified code.** 
Fully tested by compiling with MCP.pullUp(0, HIGH); checking voltage, commenting out and recompiling.

### CODE REVIEW

**Review by ladyada — APPROVED** [review]

**Comment by ladyada** [comment]
kk please check the CI errors and once passing, tag me to review!

**Comment by PsuFan** [comment]
@ladyada ugh I have a learning curve on GitHub... I don't know what the problem is. It runs fine for me, I don't know how it could verify with a MCP23017 lol

**Comment by dherrada** [comment]
@PsuFan Hey! What GitHub Actions is currently failing on is the clang-format check. You can find out how to install and run that here: https://learn.adafruit.com/the-well-automated-arduino-library/formatting-with-clang-format

If you have any questions about doing that, or you find anything confusing, let me know (please @ ping me because there's a much higher chance I'll see it if you do).

Also, don't worry about not knowing GitHub super well. We all had to learn it at some point.

Thanks!

**Comment by PsuFan** [comment]
@dherrada clang-format returns nothing...

ESP8266 Serial\RFIDDoorSystem>clang-format -i Adafruit_MCP23017.cpp

ESP8266 Serial\RFIDDoorSystem>

**Comment by dherrada** [comment]
@PsuFan Looks like you've got some trailing whitespace on lines 12, 16, and 19, although it is pretty odd that clang didn't catch that. Maybe try running it again and then just do it manually if it doesn't fix it.

**Comment by dherrada** [comment]
Yeah, it's not supposed to return anything. It just edits the files for you

**Comment by PsuFan** [comment]
Line 12, 16, 19? I've only added lines 132-138. I'm using Windows and just editing on GitHub.com, I don't have sync setup.

**Comment by dherrada** [comment]
@PsuFan Oh yah, you're definitely right. I misread the CI readout

File: Adafruit_MCP23017.cpp [hunk 1]
```diff
@@ -295,12 +305,6 @@ void Adafruit_MCP23017::setupInterruptPin(uint8_t pin, uint8_t mode) {
   updateRegisterBit(pin, HIGH, MCP23017_GPINTENA, MCP23017_GPINTENB);
 }
 
-/**
- * Disable a pin for interrupt.
- *
- * @param pin Pin to set
- *
- */
```
Comment (dherrada) [inline Adafruit_MCP23017.cpp:303]: Any reason you removed this? It's causing the Doxygen check to fail

**Comment by dherrada** [comment]
@PsuFan The good news is that the clang check is passing. See above for the doxygen (documentation) check.

**Comment by PsuFan** [comment]
clang-format removed it... Putting it back in now...

**Comment by dherrada** [comment]
@PsuFan Looks like it's passing. Good job!

**Comment by dherrada** [comment]
@ladyada Am I good to merge?

File: Adafruit_MCP23017.cpp [hunk 2]
```diff
@@ -124,10 +124,20 @@ void Adafruit_MCP23017::begin(uint8_t addr, TwoWire *theWire) {
 
   _wire->begin();
 
+  _wire->setClock(400000);
```
Comment (ladyada) [inline Adafruit_MCP23017.cpp:127]: plz remove

**Review by ladyada — CHANGES_REQUESTED** [review]

**Comment by ladyada** [comment]
close! one request

**Comment by PsuFan** [comment]
Oops, that wasn't supposed to be in there. That actually gave me increased bandwidth when reading RFID reader!

**Review by ladyada — APPROVED** [review]

**Comment by PsuFan** [comment]
thanks! okay I have one more in examples :S

**Pull request merged**